skins/cat17/src/app/lib/numeric_input_handler.js

```diff
@@ -2,47 +2,63 @@
 
 var objectAssign = require( 'object-assign' ),
 	_ = require( 'underscore' ),
+	KEYCODE_DELIMITER_MAP = {
+		188: ',',
+		190: '.'
+	},
 	Handler = {
 		input: null,
+		delimiter: '.',
 		handle: function ( event ) {
 			var $element = this.input,
 				// @todo Revise keyCode
 				// @see https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent
 				// @see https://stackoverflow.com/a/41656511
 				keyCode = event.keyCode || event.which,
-				keysAllowed = [44, 46, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 0, 8, 9, 13];
+				keysAllowed = [
+					// numbers
+					48, 49, 50, 51, 52, 53, 54, 55, 56, 57,
+					// editing and navigation
+					0, 8, 9, 13, 37, 39, 46,
+					// Delimiters (regular & numpad)
+					188, 190, 110
+				];
 
 			// @todo Shouldn't this test for `|| event.ctrlKey !== false` instead?
 			if( !_.contains( keysAllowed, keyCode ) && event.ctrlKey === false ) {
 				event.preventDefault();
 			}
 
-			// @todo Inject delimiter (depending on locale)
-			if( ( keyCode === 44 || keyCode === 46 ) && $element.val().indexOf( '.' ) > 0 ) {
-				event.preventDefault();
+			if( !this.keyCodeIsDelimiter( keyCode ) ) {
+				return;
 			}
 
-			if( keyCode === 44 ) {
-				// @todo There must be a more elegant way
-				setTimeout(
-					function () {
-						$element.val(
-							$element.val().replace( ',', '.' )
-						);
-					}, 10
-				);
+			if (this.keyCodeIsOtherDelimiter( keyCode) || this.inputHasDelimiter() ) {
+				event.preventDefault();
 			}
+		},
+		keyCodeIsDelimiter: function ( keyCode ) {
+			return _.has( KEYCODE_DELIMITER_MAP, keyCode ) ||
+				keyCode === 110; // numpad
```

Comment:
Is there a way to calculate the numpad code from the original (injected) code? Probably not. But we probably have to create a mapping of delimiter to key, so we may as well have a 1:n mapping there.